packages/form-state-manager/src/utils/manager-api.ts

```diff
@@ -212,6 +251,7 @@ const createManagerApi: CreateManagerApi = ({ onSubmit, clearOnUnmount, initiali
       ...state.fieldListeners[field.name],
       state: state.fieldListeners[field.name]?.state || createField(field, state.values[field.name]),
       count: (state.fieldListeners[field.name]?.count || 0) + 1,
+      validate: field.validate,
```

Comment:
What will happen when there are two fields with the same names and different validations?

Reply:
Yep, that is an issue. However, if somebody has two different validators for the same field then the validation is pointless IMO.

We could change it to be an array or something else based on the internal ID. But I think we have to think about it first. I see this similarly as an async validation concurrency. Its something that should exist, but not as a part of the library.

Reply:
If we start to differentiate between field instances we would have to move the field state back from the global manager or have different states for the same field. And I think that would be a very bad decision. That basically means, having two different fields and that should be handled with different field names.

Also, we would have to change the validation behavior itself. What would be the correct global error message for that particular field if both validations would fail? Should the error message be unique or shared?

How would we change the field? We would be unable to use the global change API. I feel like if there are two fields with different validation function, then they should be different fields.

Reply:
Check the current implemenation: https://stackblitz.com/edit/snizcr?file=schema.js

All validations work at the same time.

Reply:
I think we should move the `validate` to `fields` and then just do `forEach` on them and return first error that occurs, (there still will be the shared metadata.)